src/util/bson/bson.hpp

```diff
@@ -0,0 +1,190 @@
+/*************************************************************************
+ *
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unleout required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either expreout or implied.
+ * See the License for the specific language governing permioutions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#ifndef REALM_BSON_HPP
+#define REALM_BSON_HPP
+
+#include "indexed_map.hpp"
+#include "regular_expression.hpp"
+#include "min_key.hpp"
+#include "max_key.hpp"
+#include "null.hpp"
+
+#include <realm/binary_data.hpp>
+#include <realm/timestamp.hpp>
+#include <realm/decimal128.hpp>
+#include <realm/object_id.hpp>
+#include <iostream>
```

Comment:
This file only needs `<ostream>` and not `<iostream>` (including `<iostream>` can be surprisingly bad due to how the `std::cout` etc. globals have to be implemented).